mp4.go

```diff
@@ -148,10 +582,15 @@ func (m metadataMP4) readAtomData(r io.ReadSeeker, name string, size uint32, pro
 
 		// "data" + size (4 bytes each)
 		b = b[8:]
-
 		if len(b) < 4 {
 			return fmt.Errorf("invalid encoding: expected at least %d bytes, for class, got %d", 4, len(b))
 		}
+
+		if name == "gnre" {
+			m.data[name] = getInt(b[len(b)-1:])
```

Comment:
Jumping over something here?

Reply:
The genre identifier in the "gnre" tag is stored at the end of the byte array, subsequent processing is redundant, we only need id

Reply:
Ah ok.  But then `b[len(b)-1:]` is going to be 1 `byte`, so the number will only ever be between 0 and 255 (which will not hit any of the genres added).  Is this is in the spec somewhere?

Reply:
Yes, the identifier is stored in one byte.
Sorry, couldn't find an explanation / specification.
I can provide an example.